net/node/transactionPool.go

```diff
@@ -14,88 +16,365 @@ import (
+	txnPool.initTxProcessList()
+	txnPool.initTXNPendingList()
+	txnPool.initInputUTXOList()
 }
 
 func (txnPool *TXNPool) AppendTxnPool(txn *transaction.Transaction) bool {
-	hash := txn.Hash()
-	// TODO: Call VerifyTransactionWithTxPool to verify tx
+	//verify transaction with Concurrency
+	if err, errCode := va.VerifyTransactionCanConcurrency(txn); err != nil {
+		if errCode != errors.ErrDuplicatedTx {
+			txnPool.appendTXNPendingList(txn, errCode)
+		}
+		return false
+	}
+
+	//verify transaction by pool with lock
 	txnPool.Lock()
-	txnPool.list[hash] = txn
-	txnPool.txnCnt++
-	txnPool.Unlock()
+	defer txnPool.Unlock()
+	if ok := txnPool.verifyTransactionWithTxPool(txn); !ok {
+		return false
+	}
+	txnPool.appendToProcessList(txn)
 	return true
 }
 
 // Attention: clean the trasaction Pool after the consensus confirmed all of the transcation
 func (txnPool *TXNPool) GetTxnPool(cleanPool bool) map[common.Uint256]*transaction.Transaction {
 	txnPool.Lock()
 	defer txnPool.Unlock()
+	txList := txnPool.getProcessTxnList(cleanPool)
+	return txList
+}
 
-	list := txnPool.list
-	if cleanPool == true {
-		txnPool.init()
+//Attention: clean the trasaction Pool with committed transactions.
+func (txnPool *TXNPool) CleanSubmittedTransactions(block *ledger.Block) error {
+	txnPool.Lock()
+	defer txnPool.Unlock()
+	log.Trace()
+
+	var wg sync.WaitGroup
+	wg.Add(3)
+	go func() {
+		txnPool.cleanProcessTxnPool(block.Transactions)
+		wg.Done()
+	}()
+	go func() {
+		txnPool.cleanTxnPoolUtxoMap(block.Transactions)
+		wg.Done()
+	}()
+	go func() {
+		txnPool.cleanSubmittedTransactionsOfTXNPendingList(block.Transactions)
+		wg.Done()
+	}()
+	wg.Wait()
+	txnPool.clearAndReverifyAssetIssueAmoutSummaryMap()
+	return nil
+}
+
+func (txnPool *TXNPool) GetTransaction(hash common.Uint256) *transaction.Transaction {
+	txnPool.RLock()
+	defer txnPool.RUnlock()
+	return txnPool.getTransaction(hash)
+}
+
+func (txnPool *TXNPool) ClearTransactionsPendingPool() {
+	txnPool.Lock()
+	defer txnPool.Unlock()
+	txnPool.clearTransactionsPendingPool()
+	return
+}
+
+func (txnPool *TXNPool) GetTransactionPendingReason(hash common.Uint256) errors.ErrCode {
+	return txnPool.getTransactionPendingReason(hash)
+}
+
+func (node *node) SynchronizeTxnPool() {
+	node.nbrNodes.RLock()
+	defer node.nbrNodes.RUnlock()
+
+	for _, n := range node.nbrNodes.List {
+		if n.state == ESTABLISH {
+			msg.ReqTxnPool(n)
+		}
 	}
-	return DeepCopy(list)
 }
 
-func DeepCopy(mapIn map[common.Uint256]*transaction.Transaction) map[common.Uint256]*transaction.Transaction {
-	reply := make(map[common.Uint256]*transaction.Transaction)
-	for k, v := range mapIn {
-		reply[k] = v
+func (txnPool *TXNPool) clearAndReverifyAssetIssueAmoutSummaryMap() {
+	txnPool.clearAssetIssueAmoutSummaryMap()
+	txnPool.reverifyIssueAmoutInPool(txnPool.getTxList())
+
+}
+
+func (txnPool *TXNPool) verifyTransactionWithTxPool(txn *transaction.Transaction) bool {
+	//check weather have duplicate UTXO input
```

Comment:
What's the benefit  compare with verification transaction in one by one Vs verification the TXN in three different map. Any obvious performance improving in currently case?